pkg/scheduler/framework/framework_test.go

```diff
@@ -2740,3 +2740,228 @@ func TestEqualDecisions(t *testing.T) {
+					name: dummyPostBatchPluginNameB,
+					postBatchRunner: func(ctx context.Context, state CycleStatePluginReadWriter, policy *fleetv1beta1.ClusterSchedulingPolicySnapshot) (size int, status *Status) {
+						return 1, nil
+					},
+				},
+			},
+			desiredBatchSize: 10,
+			wantBatchLimit:   1,
+		},
+		{
+			name: "multple plugins, one success, one error",
+			postBatchPlugins: []PostBatchPlugin{
+				&DummyAllPurposePlugin{
+					name: dummyPostBatchPluginNameA,
+					postBatchRunner: func(ctx context.Context, state CycleStatePluginReadWriter, policy *fleetv1beta1.ClusterSchedulingPolicySnapshot) (size int, status *Status) {
+						return 0, FromError(fmt.Errorf("internal error"), dummyPostBatchPluginNameA)
+					},
+				},
+				&DummyAllPurposePlugin{
+					name: dummyPostBatchPluginNameB,
+					postBatchRunner: func(ctx context.Context, state CycleStatePluginReadWriter, policy *fleetv1beta1.ClusterSchedulingPolicySnapshot) (size int, status *Status) {
+						return 1, nil
+					},
+				},
+			},
+			desiredBatchSize: 10,
+			wantStatus:       FromError(fmt.Errorf("internal error"), dummyPostBatchPluginNameA),
+		},
+		{
+			name: "single plugin, skip",
+			postBatchPlugins: []PostBatchPlugin{
+				&DummyAllPurposePlugin{
+					name: dummyPostBatchPluginNameA,
+					postBatchRunner: func(ctx context.Context, state CycleStatePluginReadWriter, policy *fleetv1beta1.ClusterSchedulingPolicySnapshot) (size int, status *Status) {
+						return 0, NewNonErrorStatus(Skip, dummyPostBatchPluginNameA)
+					},
+				},
+			},
+			desiredBatchSize: 10,
+			wantBatchLimit:   10,
+		},
+		{
+			name: "single plugin, unschedulable",
+			postBatchPlugins: []PostBatchPlugin{
+				&DummyAllPurposePlugin{
+					name: dummyPostBatchPluginNameA,
+					postBatchRunner: func(ctx context.Context, state CycleStatePluginReadWriter, policy *fleetv1beta1.ClusterSchedulingPolicySnapshot) (size int, status *Status) {
+						return 1, NewNonErrorStatus(ClusterUnschedulable, dummyPostBatchPluginNameA)
+					},
+				},
+			},
+			desiredBatchSize: 10,
+			wantStatus:       FromError(fmt.Errorf("internal error"), dummyPostBatchPluginNameA),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			profile := NewProfile(dummyProfileName)
+			for _, p := range tc.postBatchPlugins {
+				profile.WithPostBatchPlugin(p)
+			}
+			f := &framework{
+				profile: profile,
+			}
+
+			ctx := context.Background()
+			state := NewCycleState()
+			state.desiredBatchSize = tc.desiredBatchSize
+			policy := &fleetv1beta1.ClusterSchedulingPolicySnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: policyName,
+				},
+			}
+			batchLimit, status := f.runPostBatchPlugins(ctx, state, policy)
+			if batchLimit != tc.wantBatchLimit {
+				t.Errorf("runPostBatchPlugins() batch limit = %d, want %d", batchLimit, tc.wantBatchLimit)
+			}
+			if diff := cmp.Diff(status, tc.wantStatus, cmpopts.IgnoreUnexported(Status{}), ignoredStatusFields); diff != "" {
```

Comment:
sorry, missed a comment before.

We need to use the these two options here.
cmpopts.IgnoreFields(Status{}, "reasons", "err") & cmp.AllowUnexported(Status{})

otherwise, cmp will ignore the unexported fields.